user_test.go

```diff
@@ -19,6 +20,92 @@ func TestGetUserInfo(t *testing.T) {
 	require.Nil(t, err)
 	assert.Equal(t, PassageUserID, user.ID)
 }
+
+func TestGetUserInfoByIdentifier(t *testing.T) {
+	psg, err := passage.New(PassageAppID, &passage.Config{
+		APIKey: PassageApiKey,
+	})
+	require.Nil(t, err)
+
+	createUserBody := passage.CreateUserBody{
+		Email: RandomEmail,
+	}
+
+	user, err := psg.CreateUser(createUserBody)
+	require.Nil(t, err)
+	assert.Equal(t, RandomEmail, user.Email)
+
+	userByIdentifier, err := psg.GetUserByIdentifier(RandomEmail)
+	require.Nil(t, err)
+
+	userById, err := psg.GetUser(user.ID)
+	require.Nil(t, err)
+
+	assert.Equal(t, user.ID, userById.ID)
+
+	assert.Equal(t, userById, userByIdentifier)
+}
+
+func TestGetUserInfoByIdentifierEmailUpperCase(t *testing.T) {
+	psg, err := passage.New(PassageAppID, &passage.Config{
+		APIKey: PassageApiKey,
+	})
+	require.Nil(t, err)
+
+	email := strings.ToUpper(RandomEmail)
```

Comment:
nitpick: creating the user with an uppercase email and then searching for it with the same value wouldn't necessarily prove that the search is case insensitive. If the `ToUpper` happened after `CreateUser` and before or at `GetUserByIdentifier` instead, it would be more accurate since this client doesn't know that the email is lowercased in the server.

Reply:
Good catch! They are not suppose to be the same value! Updating this